sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -0,0 +1,685 @@
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+
+  peripheral_init();
+
+  // Enable all interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+
+  // endable alert info
+  CHECK_DIF_OK(dif_rstmgr_alert_info_set_enabled(&rstmgr, kDifToggleEnabled));
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rst_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  LOG_INFO("reset info = 0x%02X", rst_info);
+  global_alert_called = 0;
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    global_test_round = kRound1;
+    prgm_alert_handler_round1();
+
+    LOG_INFO("ALERT1");
+    CHECK_DIF_OK(dif_i2c_alert_force(&i2c0, kDifI2cAlertFatalFault));
+    CHECK_DIF_OK(dif_i2c_alert_force(&i2c1, kDifI2cAlertFatalFault));
+    CHECK_DIF_OK(dif_i2c_alert_force(&i2c2, kDifI2cAlertFatalFault));
+    CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+        &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+    // Give a enough delay until sw rest happens.
+    LOG_INFO("wait round1 100us");
+    busy_spin_micros(100);
+
+  } else if (rst_info == kDifRstmgrResetInfoSw) {
+    collect_alert_dump_and_compare(kRound1);
+    global_test_round = kRound2;
+
+    prgm_alert_handler_round2();
+
+    LOG_INFO("ALERT2");
+    // Setup aon_timer the wdog bark and bite timeouts.
+    uint64_t bark_cycles =
+        udiv64_slow(kWdogBarkMicros * kClockFreqAonHz, 1000000, NULL);
+    uint64_t bite_cycles =
+        udiv64_slow(kWdogBiteMicros * kClockFreqAonHz, 1000000, NULL);
+    aon_timer_testutils_watchdog_config(&aon_timer, bark_cycles, bite_cycles,
+                                        false);
+
+    CHECK_DIF_OK(dif_uart_alert_force(&uart0, kDifUartAlertFatalFault));
+    CHECK_DIF_OK(dif_uart_alert_force(&uart1, kDifUartAlertFatalFault));
+    CHECK_DIF_OK(dif_uart_alert_force(&uart2, kDifUartAlertFatalFault));
+    CHECK_DIF_OK(dif_uart_alert_force(&uart3, kDifUartAlertFatalFault));
+    CHECK_DIF_OK(dif_otp_ctrl_alert_force(&otp_ctrl,
+                                          kDifOtpCtrlAlertFatalBusIntegError));
+    CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+        &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+    CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+        &alert_handler, kDifAlertHandlerIrqClassc, kDifToggleEnabled));
+
+    LOG_INFO("wait round2 100us");
```

Comment:
Same as for line 638, here and line 677.

Reply:
fxd